### PR TITLE
allow keep and library owners to delete comments

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner @ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 1]
```diff
@@ -219,6 +220,24 @@ class KeepCommanderImpl @Inject() (
     }
   }
 
+  def canDeleteCommentOnKeep(userId: Id[User], keepId: Id[Keep]): Boolean = {
+    db.readOnlyReplica { implicit s =>
+      keepRepo.getOption(keepId) match {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: Is this a worry? Typically, if someone has access to a `Id[Keep]`, it came as a key from the db and we _want_ to blow up if it's missing later.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 2]
```diff
@@ -219,6 +220,24 @@ class KeepCommanderImpl @Inject() (
...
+        case None =>
+          airbrake.notify(s"no keep found for id=${keepId.id}, user ${userId.id} can't delete comment")
+          false
+        case Some(keep) =>
+          keep.libraryId match {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: You may want to use `keepToLibraryRepo` here to be safer going forward.

**Comment by andrewconner** [comment]
Minor comments, otherwise :+1:

**Comment by camhashemi** [comment]
@andrewconner thanks

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 3]
```diff
@@ -201,24 +202,19 @@ class KeepCommanderImpl @Inject() (
...
+      permissionCommander.getLibraryPermissions(ktl.libraryId, Some(userId)).contains(LibraryPermission.ADD_COMMENTS)
     }
   }
 
+  def canDeleteCommentOnKeep(userId: Id[User], keepId: Id[Keep]): Boolean = db.readOnlyReplica { implicit s =>
+    keepRepo.get(keepId).userId == userId ||
+      libraryRepo.getActiveByIds(ktlRepo.getAllByKeepId(keepId).map(_.libraryId).toSet).exists {
+        case (_, library) => library.ownerId == userId
+      }
+  }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:216]: @andrewconner @ryanpbrewster going with the strict permissions route for adding, not worried about deletion (plus it's impossible to distinguish between library owners)

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala:216]: Works for me

**Pull request merged**